src/models/game.py

```diff
@@ -20,15 +45,56 @@ class Game:
     current_player_id: t.Optional[int]
     step_count: int
     winner_id: t.Optional[int]
-    field: t.Union[str, list]
+    field: t.Union[str, t.List[chr]]
     current_state: str
+    owner_mark: chr = 'X'
+    opponent_mark: chr = '0'
+
+    def _game_loop(self, field: t.List[str]):
+        for condition in WIN_CONDITIONS:
+            if all(field[cell_id] == self._current_mark() for cell_id in condition):
+                self.winner_id = self.current_player_id
+                self.current_state = State.DONE.value
+                break
+
+    def _check_win(self):
+        one_line_field = [colum for line in self.field for colum in line]
+        if one_line_field.count('') == 0:
+            self.current_state = State.DONE.value
+        else:
+            self._game_loop(one_line_field)
+
+    def _invert_player(self):
+        if self.current_player_id == self.owner_id:
+            self.current_player_id = self.opponent_id
+        else:
+            self.current_player_id = self.owner_id
+
+    def _current_mark(self) -> chr:
+        if self.current_player_id == self.owner_id:
+            return self.owner_mark
+        return self.opponent_mark
 
     def set_opponent(self, opponent: User):
         self.opponent_id = opponent.id
         self.current_player_id = random.choice(
             [self.owner_id, self.opponent_id]
         )
+        self.owner_mark, self.opponent_mark = 'X', '0'
         self.current_state = State.IN_GAME.value
 
+    def update(self, coords: t.List[int]):
+        if self.field[coords[0]][coords[1]] not in ('X', '0'):
+            field_size = len(self.field) ** 2
+            self.field[coords[0]][coords[1]] = self._current_mark()
+            self.step_count += 1
+            if self.step_count >= math.ceil(field_size / 2):
+                self._check_win()
+            self._invert_player()
+            if self.step_count > field_size:
```

Comment:
step_count cant be > than field_size, if I understand right, max value of step count - is value of cels in field, currently is 9, so maximum value of step_count is 9, and 9 not greater than 9, so here must be >= I guess